File: backend/src/main/java/com/fhsocial/backend/Enums/SseType.java
package com.fhsocial.backend.Enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum SseType {
    ADD_USER("addUser"),
    REMOVE_USER("removeUser"),
    ADD_EVENT("addEvent"),
    REMOVE_EVENT("removeEvent"),  
    ADD_MEMBER("addMember"),
    REMOVE_MEMBER("removeMember"),
    ADD_COMMENT("addComment"),
    REMOVE_COMMENT("removeComment"),
    ADD_FILE_PREVIEW("addFilePreview"),
    REMOVE_FILE_PREVIEW("removeFilePreview");

    private final String value;

    SseType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }
}

File: frontend/lib/enums/sse_type.dart

enum SseType {
  addEvent('addEvent'),
  removeEvent('removeEvent'),  
  addMember('addMember'),
  removeMember('removeMember'),
  addComment('addComment'),
  removeComment('removeComment'),
  addFilePreview('addFilePreview'),
  removeFilePreview('removeFilePreview');

  const SseType(this.value);

  final String value;

  static SseType? parse(String value) {
    for (final type in SseType.values) {
      if (type.value == value) {
        return type;
      }
    }
    return null;
  }
}